### PR TITLE
add a message why one should not use GridRefinement functions with co…

### PR DESCRIPTION
**Pull request by davydden**
…mplex-valued PETSc

fixes https://github.com/dealii/dealii/issues/2255

In reference to https://github.com/dealii/dealii/issues/2033 .

### CODE REVIEW

**Comment by davydden** [comment]
i did not check that it compiles... doing it now...

**Comment by davydden** [comment]
all good.

File: source/grid/grid_refinement.cc [hunk 1]
```diff
@@ -75,7 +75,8 @@ namespace
       for (unsigned int i=0; i<criteria.size(); ++i)
         m = std::max (m, criteria(i));
 #else
-      Assert(false, ExcNotImplemented())
+      Assert(false, ExcMessage("The GridRefinement functions should only get real-valued vectors of refinement indicators."
+                               "Using these functions with complex-valued PETSc vectors does not make sense."))
```
Comment (bangerth) [inline source/grid/grid_refinement.cc]: Please add a space to the beginning of this or the end of the previous line.

File: source/grid/grid_refinement.cc [hunk 2]
```diff
@@ -93,7 +94,8 @@ namespace
       for (unsigned int i=1; i<criteria.size(); ++i)
         m = std::min (m, criteria(i));
 #else
-      Assert(false, ExcNotImplemented());
+      Assert(false, ExcMessage("The GridRefinement functions should only get real-valued vectors of refinement indicators."
+                               "Using these functions with complex-valued PETSc vectors does not make sense."))
```
Comment (bangerth) [inline source/grid/grid_refinement.cc]: Same here.

**Comment by bangerth** [comment]
OK with the two minor edits.

**Pull request merged**